Client/Extensions/UIPasteboardExtensions.swift

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import UIKit
+
+extension UIPasteboard {
+
+    func addImageWithData(data: NSData, forURL url: NSURL) {
+        let isGIF = UIImage.dataIsGIF(data)
+
+        guard let image = imageWithData(data, isGIF: isGIF) else {
```

Comment:
turns out you do not need to turn the NSData into image. If you set the NSData directly with the gif key into the pasteboard then it pastes as animated.
```
let isGIF = UIImage.dataIsGIF(data)

        // Setting pasteboard.items allows us to set multiple representations for the same item.
        items = [[
            kUTTypeURL as String: url,
            imageTypeKey(isGIF): data
        ]]
```